### PR TITLE
Bump memory-db lib to 0.24.1 and add shrink_to_fit #11760 

### PR DESCRIPTION
**Pull request by rakita**
- Mostly fixes that help to resolve the issue: #11760 
- Bump `memory-db lib` to `0.24.1` and add shrink_to_fit
- Upgraded some trace log to info when pruning happens
- Use `overlay.cumulative_size` instead of `overlay.backing_overlay.size_of(..)` It is more inaccurate but it is good enough and a it will allow us to skip heavy `size_of`.

### CODE REVIEW

**Comment by rakita** [comment]
Tested for four days, everything function as expected.

File: util/journaldb/src/overlayrecentdb.rs [hunk 1]
```diff
@@ -253,7 +253,7 @@ impl JournalDB for OverlayRecentDB {
 		let mut mem = self.transaction_overlay.size_of(&mut ops);
 		let overlay = self.journal_overlay.read();
 
-		mem += overlay.backing_overlay.size_of(&mut ops);
+		mem += overlay.cumulative_size; //This does not present real HashMap memory footprint but we are doing shrink_to_size after removal so this is good approximation.
```
Comment (ordian) [inline util/journaldb/src/overlayrecentdb.rs:256]: note that memory-db 0.24 includes https://github.com/paritytech/trie/pull/94, which should not be heavy to compute

Comment (rakita) [reply util/journaldb/src/overlayrecentdb.rs:256]: Hm, it looks like it is calculated same as cumulative_size. Thanks for the note! For now, I would like to just use comulative_size so as to not change much in code but this is something that we could use in the future.

Comment (ordian) [reply util/journaldb/src/overlayrecentdb.rs:256]: No, `cumulative_size` does not keep track the `HashMap` memory footprint as you noticed in the comments, while `overlay.backing_overlay.size_of(&mut ops)` does and it does it in O(1) so should not be heavy.

Comment (rakita) [reply util/journaldb/src/overlayrecentdb.rs:256]: Sorry, it seems I missed something. I will include the change, thanks again.

**Comment by rakita** [comment]
Not sure why, but test unrelated to this pull request fails on macos: command is `cargo test -p ethstore-cli --test cli` i tried it on my windows machine and linux server and both tests pass. I dont have mac to check this

```
     Running `/Users/runner/work/openethereum/openethereum/target/release/deps/cli-608eb763b339d18b`

running 1 test
test cli_cmd ... FAILED

failures:

---- cli_cmd stdout ----
thread 'cli_cmd' panicked at 'assertion failed: output.status.success()', accounts/ethstore/cli/tests/cli.rs:29:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    cli_cmd

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '-p ethstore-cli --test cli'
```

**Comment by rakita** [comment]
Blocked. The test fails on macos because of updated memory-db and I don't have platform to debug this.

**Comment by vorot93** [comment]
Test failure is not related to this pull request.

**Pull request merged**